webapp/channels/.eslintrc.json

```diff
@@ -37,19 +37,55 @@
     "eol-last": ["error", "always"],
     "import/no-unresolved": 2,
     "import/order": [
-      0,
+      2,
       {
-        "newlines-between": "always-and-inside-groups",
+        "newlines-between": "always",
         "groups": [
           "builtin",
           "external",
-          [
-            "internal",
-            "parent"
-          ],
+          "internal",
           "sibling",
+          "parent",
           "index"
-        ]
+        ],
+        "pathGroups": [
+          {
+            "pattern": "react*{,/**}",
+            "group": "builtin",
+            "position": "before"
+          },
+          {
+            "pattern": "@mattermost/**",
+            "group": "external",
+            "position": "after"
+          },
+          {
+            "pattern": "mattermost-redux/**",
+            "group": "external",
+            "position": "after"
+          },
+          {
+            "pattern": "@(selectors|actions|stores|store|reducers){,/**}",
+            "group": "external",
+            "position": "after"
+          },
+          {
+            "pattern": "components/**",
+            "group": "external",
+            "position": "after"
+          },
+          {
+            "pattern": "types{,/**}",
+            "group": "internal",
+            "position": "after"
+          }
```

Comment:
What was your thinking for having internal types after everything else? I figured they would just be mixed with other imports, but I'm 0.5 and that's mostly just because I'm curious

Reply:
This is the types folder. If I don't do this, a lot of types will be between utils and other imports, and I felt that being messy. It is not so much about about having all the `import type` together, and more about segregating the type folder in its own group.

Reply:
Ah, okay. I guess that might be another point towards separate groups for all the top level folders, but I'm still undecided on if that's actually good since it could lead to a lot of empty lines, so this works well for me